newsfragments/2841.feature.rst

```diff
@@ -0,0 +1,3 @@
+To better reflect the underlying thread handling semantics,
+`trio.to_thread.run_sync` gained a new keyword argument ``abandon_on_cancel``
+that overrides the older (now deprecated) ``cancellable`` flag.
```

Comment:
Hmm it's not exactly deprecated. I'm not sure how to best phase "replaced temporary kwarg name" or whatever though. I feel like it's important information that `cancellable` was never released.

Reply:
I'm happy to simply not say deprecated anywhere!

Reply:
Yeah that probably works

Reply:
Oh wait gahhh I shouldn't do this while sleepy, I only saw the changes that replaced the kwarg, didn't realize you kept it for compatibility 😭 

Sorry... if we are going to keep the kwarg, then deprecation is good. I thought the function that "cancelled" is on was added in this release meaning we could just replace the name without deprecation period.

Reply:
Haha no worries, but I won't be able to revert it today. Feel free to merge either version in the mean while.